parse.go

```diff
@@ -0,0 +1,566 @@
+package dicom
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/gradienthealth/go-dicom/dicomio"
+	"github.com/gradienthealth/go-dicom/dicomlog"
+	"github.com/gradienthealth/go-dicom/dicomtag"
+)
+
+// GoDICOMImplementationClassUIDPrefix defines the UID prefix for
+// go-dicom. Provided by https://www.medicalconnections.co.uk/Free_UID
+const GoDICOMImplementationClassUIDPrefix = "1.2.826.0.1.3680043.9.7133"
+
+var GoDICOMImplementationClassUID = GoDICOMImplementationClassUIDPrefix + ".1.1"
+
+const GoDICOMImplementationVersionName = "GODICOM_1_1"
+
+// Parser represents an entity that can read and parse DICOMs
+type Parser interface {
+	// Parse DICOM data
+	Parse(options ParseOptions) (*DataSet, error)
+	// ParseNext reads and parses the next element
+	ParseNext(options ParseOptions) *Element
```

Comment:
From before when this was `ReadElement` it has not returned an error, rather it sets an error on the `dicomio.Decoder`...I expect this was done in the past because it is convenient to just have a `*Element` returned and deal with the errors later (particularly given there seem to be recursive calls to this function)...should see if it makes sense to have this return a canonical `error` though